### ISSUE TITLE
Implement #Bool/induction in impredicative setoid encoding

### ISSUE THREAD

**Issue by nponeccop** [groupoid/om#25]
Basically, a function from proofs of `P true` and `P false` to a proof of `forall x : Bool . P x`

**Comment by nponeccop** [comment]
See http://nponeccop.livejournal.com/480539.html but there are questons

**Comment by nponeccop** [comment]
It turned out that "normal" Bool eliminator is impossible. So this task becomes essentially the task of implementing the impredicative setoid-based encoding (which didn't exist before as a separate task).

Blocked by #30

**Comment by 5HT** [comment]
$ time (./om print type a "#Data/Bool/induc" , mode "posets" > /dev/null)

real	0m1.055s
user	0m0.962s
sys	0m0.291s
